.github/workflows/release.yml

```diff
@@ -425,6 +433,24 @@ jobs:
                   cd -
               fi
           done
+
+      - name: "Run smoke-tests"
+
+        # This job only runs whenever a tag is created. A tag is required
+        # for a functional plugin compiler build for when the GO_GET=1 env
+        # is provided. The plugin compiler cannot fetch the referenced
+        # commit from a PR, but requires a /heads or /tags reference.
+        #
+        # See https://github.com/golang/go/issues/31191 for more info.
+
+        if: startsWith(github.ref, 'refs/tags')
```

Comment:
Now that we build images and push them in PRs, we should certainly run these in the PR context.

Reply:
We can't due to the caveat listed above (no way to `go get` from a PR commit, it needs to be in heads or tags)